kobo/apps/superuser_stats/tasks.py

```diff
@@ -1,21 +1,87 @@
 # coding: utf-8
+import unicodecsv
+
 from celery import shared_task
 from django.conf import settings
+from django.core.files.storage import get_storage_class
+
+from hub.models import ExtraUserDetail
+from kobo.static_lists import COUNTRIES
+from kpi.constants import ASSET_TYPE_SURVEY
+from kpi.deployment_backends.kc_access.shadow_models import (
+    KobocatUser,
+    KobocatUserProfile,
+    KobocatXForm,
+    ReadOnlyKobocatInstance,
+)
+from kpi.models.asset import Asset
 
 # Make sure this app is listed in `INSTALLED_APPS`; otherwise, Celery will
 # complain that the task is unregistered
 
 
+@shared_task
+def generate_country_report(output_filename, start_date, end_date):
+
+    def format_date(d):
+        if hasattr(d, 'strftime'):
+            return d.strftime('%F')
+        else:
+            return d
+
+    def get_row_for_country(code, label):
```

Comment:
Would you mind to add function annotations. (Feel free to add them to existing code in this file too). The signature of the inner function is quite simple but it helps to find out which type each variable is pretty quickly.

Moreover, as a side note: I'm used to use different variable names inside inner functions to avoid shadowing  variables in the same scope.   

For example: 
```python
def generate_country_report(output_filename: str, start_date: date, end_date: date):
    ... 
    def get_row_for_country(code_: str, label_: str) -> dict:   # use `code_` instead of `code` and `label_` instead  of `label` 
        row_ = []   # use `row_` instead of `row`
        ...
        return row_
        
        
    with default_storage.open(output_filename, 'wb') as output_file:
        ...
        for code, label in COUNTRIES:

            try:
                row = get_row_for_country(code, label)  # `row`, `code, `label` are not shadowed by `get_row_for_country()` variables. 
            except Exception as e:
                row = ['!FAILED!', 'Country: {}'.format(label), repr(e)]
            writer.writerow(row)     
```            

It's not a big deal, the code works the way it is, but IMHO, it avoids making some mistake and it is more obvious when you debug to find out the value of each variable.